src/ophys_etl/transforms/create_slapp_inputs.py

```diff
@@ -0,0 +1,270 @@
+import argschema
+import json
+import numpy as np
+from typing import List, Any
+from pathlib import Path
+import marshmallow as mm
+from marshmallow import ValidationError
+
+from ophys_etl.schemas.fields import H5InputFile
+
+
+class ExperimentSchema(argschema.schemas.DefaultSchema):
+    experiment_id = argschema.fields.Int(
+        required=True,
+        description="experiment id")
+    local_ids = argschema.fields.List(
+        argschema.fields.Int,
+        required=False,
+        cli_as_single_argument=True,
+        description="which ROIs to select from this experiment")
+    binarized_rois_path = argschema.fields.InputFile(
+        required=False,
+        description="path to output from ophys_etl.convert_rois")
+    traces_h5_path = H5InputFile(
+        required=False,
+        description="path to traces file output from extract_traces")
+    movie_path = H5InputFile(
+        required=True,
+        description="path to mtion corrected movie")
+
+
+class SlappTransformInputSchema(argschema.ArgSchema):
+    global_id_offset = argschema.fields.Int(
+        required=True,
+        description="global ROI ids will start at this value")
+    experiments = argschema.fields.Nested(ExperimentSchema, many=True)
+    random_seed = argschema.fields.Int(
+        required=True,
+        default=42,
+        description=("seed for random number generator to select ROIs. "
+                     "ignored if n_roi_total is not set."))
+    n_roi_total = argschema.fields.Int(
+        required=False,
+        description=("if provided, experiments.local_ids will be ignored "
```

Comment:
Would be helpful for a naive user to explain a reason they might provide this value (e.g. constraining the input size of the labeling job)

Reply:
done